examples/neural_networks/plot_mlp_training_curves.py

```diff
@@ -52,23 +57,29 @@ def plot_on_dataset(X, y, ax, name):
     # for each dataset, plot learning for each learning strategy
     print("\nlearning on dataset %s" % name)
     ax.set_title(name)
-    X = MinMaxScaler().fit_transform(X)
-    mlps = []
-    if name == "digits":
-        # digits is larger but converges fairly quickly
-        max_iter = 15
-    else:
-        max_iter = 400
-
-    for label, param in zip(labels, params):
-        print("training: %s" % label)
-        mlp = MLPClassifier(verbose=0, random_state=0,
-                            max_iter=max_iter, **param)
-        mlp.fit(X, y)
-        mlps.append(mlp)
-        print("Training set score: %f" % mlp.score(X, y))
-        print("Training set loss: %f" % mlp.loss_)
-    for mlp, label, args in zip(mlps, labels, plot_args):
+
+    # some parameter combinations will not converge as can be seen on the
+    # plots so they are ignored here
+    with warnings.catch_warnings():
+        warnings.filterwarnings("ignore", category=ConvergenceWarning,
+                                module="sklearn")
+        X = MinMaxScaler().fit_transform(X)
+        mlps = []
+        if name == "digits":
+            # digits is larger but converges fairly quickly
+            max_iter = 15
+        else:
+            max_iter = 400
+
+        for label, param in zip(labels, params):
+            print("training: %s" % label)
+            mlp = MLPClassifier(verbose=0, random_state=0,
+                                max_iter=max_iter, **param)
+            mlp.fit(X, y)
```

Comment:
I think the context manager should just be for this line

Reply:
I was putting it before preprocessor instead of the model fitting. Sorry for that.

Reply:
But should it be before the `mpl.fit` or where I define the model?